src/object/cli_obj.c

```diff
@@ -4839,15 +4869,18 @@ obj_retry_next_shard(struct dc_object *obj, struct obj_auxi_args *obj_auxi,
 	*shard = (obj_auxi->req_tgts.ort_shard_tgts[0].st_shard + 1) % grp_size + start_shard;
 	if (*shard == obj_auxi->initial_shard) {
 		obj_auxi->no_retry = 1;
-		if (obj_auxi->csum_retry)
-			return -DER_CSUM;
-
-		return -DER_TX_UNCERTAIN;
+		return retry_errcode(obj_auxi, 0);
 	}
 
 	return rc;
 }
 
+static inline bool
+obj_auxi_retryable(struct obj_auxi_args *obj_auxi)
```

Comment:
"retryable" is some misguided in client IO logic, because client will retry the IO request under many cases, for example: RPC timeout, DER_INPROGRESS, stale pool map, and the cases you mention here.

In your case, the retry is for next replica, so we need a more proper name, such as "obj_auxi_need_retry_redudancy"?

Reply:
Sure.